ttnctl/cmd/downlink.go

```diff
@@ -36,71 +36,64 @@ var downlinkCmd = &cobra.Command{
 		}
 		ctx = ctx.WithField("DevID", devID)
 
-		JSON, err := cmd.Flags().GetBool("json")
+		jsonFlag, err := cmd.Flags().GetBool("json")
 
 		if err != nil {
-			ctx.WithError(err).Fatal("Error: value of the json flag")
+			ctx.WithError(err).Fatal("Failed to read json flag")
 		}
 
-		if JSON {
-			fields := args[1]
-			if fields == "" {
-				ctx.WithError(err).Fatal("No fields or payload provided")
-			}
-
-			// Valid payload provided + json flag
-			_, err := types.ParseHEX(args[1], len(args[1])/2)
-			if err == nil {
-				ctx.WithError(err).Fatal("You are providing a valid payload using the --json flag.")
-			}
+		fPort, err := cmd.Flags().GetInt("fport")
 
-			fPort, _ := cmd.Flags().GetInt("fport")
+		if err != nil {
+			ctx.WithError(err).Fatal("Failed to read port flag")
+		}
 
-			token := client.PublishDownlink(mqtt.DownlinkMessage{
-				AppID:  appID,
-				DevID:  devID,
-				FPort:  uint8(fPort),
-				Fields: parseJSON(fields),
-			})
-			token.Wait()
-			if token.Error() != nil {
-				ctx.WithError(token.Error()).Fatal("Could not enqueue downlink")
-			}
+		message := mqtt.DownlinkMessage{
+			AppID: appID,
+			DevID: devID,
+			FPort: uint8(fPort),
+		}
 
-			ctx.Info("Enqueued downlink")
-		} else { // Payload provided
-			payload, err := types.ParseHEX(args[1], len(args[1])/2)
-			if err != nil {
-				ctx.WithError(err).Fatal("Invalid Payload")
+		if args[1] != "" {
```

Comment:
To be sure that you provide something at :

ttnctl downlink <something>

If we have something then we analyze it, if not, we return the command usage to the user to tell him